testcontainers/src/clients/http.rs

```diff
@@ -411,49 +402,55 @@ mod tests {
 
     #[tokio::test(flavor = "multi_thread")]
     async fn http_run_command_should_include_name() {
-        let docker = Http::new();
+        let docker = docker_client();
         let image = GenericImage::new("hello-world", "latest");
-        let image = RunnableImage::from(image).with_container_name("hello_container");
-        let container = docker.run(image).await;
+        let container = RunnableImage::from(image)
+            .with_container_name("hello_container")
```

Comment:
Can we figure out a way to make this more ergonomic? Ideally, I'd like users to never need to construct `RunnableImage` but only `Image`s. `with_container_name` is defined in a trait, right? Does it already work to just write:

```rust
GenericImage::new().with_container_name().start().await;
```